lib/cron/cache-pools-ipfs.ts

```diff
@@ -0,0 +1,112 @@
+import pinataSDK from '@pinata/sdk';
+import { ChainId, SubgraphProvider } from '@uniswap/smart-order-router';
+import { EventBridgeEvent, ScheduledHandler } from 'aws-lambda';
+import { Route53, STS } from 'aws-sdk';
+import { default as bunyan, default as Logger } from 'bunyan';
+import fs from 'fs';
+
+// if (process.env.STAGE == 'beta') {}
+
+const DIRECTORY = '/tmp/pools/';
+
+// add more chains here
+const chains = [ChainId.MAINNET, ChainId.RINKEBY]
+const names = ["mainnet.txt", "rinkeby.txt"]
+
+const pinata = pinataSDK(
+  process.env.PINATA_API_KEY!,
+  process.env.PINATA_API_SECRET!
+);
+
+const handler: ScheduledHandler = async (
+  event: EventBridgeEvent<string, void>
+) => {
+  const log: Logger = bunyan.createLogger({
+    name: 'RoutingLambda',
+    serializers: bunyan.stdSerializers,
+    level: 'info',
+    requestId: event.id,
+  });
+
+  const sts = new STS();
+  const stsParams = {
+    RoleArn: process.env.ROLE_ARN!,
+    RoleSessionName: `UpdateApiRoute53Role`,
+  };
+
+  // init route53 with credentials
+  let credentials;
+  var route53 = new Route53();
+  sts.assumeRole(stsParams, (err: any, data: any) => {
+    if (err) {
+      log.error({ err }, `Error assuming role`);
+    } else {
+      credentials = data.Credentials;
+      route53 = new Route53({ credentials: credentials });
+      log.info(`Role assumed`);
+    }
+  });
+
+  for (let i = 0; i < chains.length; i ++ ) {
+    const subgraphProvider = new SubgraphProvider(chains[i], 3, 15000);
+    const pools = await subgraphProvider.getPools();
+    // TODO : filter pools 
+    const poolString = JSON.stringify(pools);
+
+    let fileName = names[i]
+
+    // create directory and file
+    fs.mkdirSync(DIRECTORY, { recursive: true });
+    fs.writeFileSync(DIRECTORY.concat(fileName), poolString);
```

Comment:
Nit: best practice to use Node's `path` module for all things directory/path related as it handles awkward things that cause bugs like extra slashes and works cross platform. 

`path.join` would be good here: https://nodejs.org/api/path.html#path_path_join_paths